core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -468,12 +479,12 @@ object Battle {
         val capturedUnitTile = capturedUnit.getTile()
 
         when {
-            // Uncapturable units are destroyed (units captured by barbarians also - for now)
-            defender.unit.hasUnique("Uncapturable") || attacker.getCivInfo().isBarbarian() -> {
+            // Uncapturable units are destroyed
+            defender.unit.hasUnique("Uncapturable") -> {
                 capturedUnit.destroy()
             }
-            // Captured settlers are converted to workers.
-            capturedUnit.name == Constants.settler -> {
+            // Captured settlers are converted to workers unless captured by barbarians (so they can be returned later).
```

Comment:
But if you recapture your settler from a barbarian, it would still turn into a worker, right?

Reply:
Yes, for now. A future PR will add memory to civilian units so that you can return them their original owner when defeating the barbarians. A settler returned in this way does not become a worker.